src/features/navigator/NarrativeControl/Delete.tsx

```diff
@@ -1,27 +1,66 @@
 /* NarrativeControl/Delete */
-import { FC } from 'react';
+import { FC, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import toast from 'react-hot-toast';
 import { Button } from '../../../common/components';
-import { useAppDispatch } from '../../../common/hooks';
-import { TODOAddLoadingState } from '../common';
-import { deleteNarrative } from '../navigatorSlice';
+import { useAppDispatch, useAppSelector } from '../../../common/hooks';
+import { isKBaseBaseQueryError } from '../../../common/api/utils/kbaseBaseQuery';
+import { deleteWorkspace } from '../../../common/api/workspaceApi';
+import {
+  generatePathWithSearchParams,
+  getParams,
+} from '../../../features/params/paramsSlice';
+import { deleteNarrative, loading, setLoading } from '../navigatorSlice';
 import { ControlProps } from './common';
 
 export const Delete: FC<ControlProps> = ({ narrativeDoc, modalClose }) => {
   const dispatch = useAppDispatch();
+  const loadState = useAppSelector(loading);
+  const params = useAppSelector(getParams);
+  const navigate = useNavigate();
+  const [userConfirmation, setUserConfirmation] = useState(false);
+  const [deleteTrigger] = deleteWorkspace.useMutation();
+
+  const wsId = narrativeDoc.access_group;
+  useEffect(() => {
+    if (loadState) return;
+    if (!userConfirmation) return;
+  });
+
+  const message = `Deleted narrative ${wsId}.`;
   const deleteNarrativeHandler = async () => {
-    await TODOAddLoadingState();
-    dispatch(deleteNarrative({ wsId: narrativeDoc.access_group }));
+    setUserConfirmation(true);
     modalClose();
+    dispatch(deleteNarrative({ wsId }));
+    try {
```

Comment:
Use and/or modify this func to solve this section? https://github.com/kbase/ui/blob/8312ed5485651ff3c76ffd1bcd9c03d72552e692/src/common/api/utils/parseError.ts#L4

Reply:
Sure!